### PR TITLE
DM-27453: Add command-line command `explog create-table`

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    ----------
+    port
+        Port on which to run the explog service.
+    message_id
+        Expected ID of the added message.
+    """
+
+    cmdline_args = ["explog", "run"]
+    if port is None:
+        port = 8080
```
Comment (SimonKrughoff) [inline tests/test_cli.py:80]: I'm curious about this.  Why not just put 8080 in the test case above?  I'm also curious why you have two ports.  Presumably one is postgres, what is the other?

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            await asyncio.wait_for(
+                run_process.wait(), timeout=TERMINATE_TIMEOUT
+            )
+        else:
+            try:
+                stderr_bytes = await run_process.stderr.read()  # type: ignore
+                stderr_msg = stderr_bytes.decode()
+            except Exception as e:
+                stderr_msg = f"could not read stderr: {e}"
+            raise AssertionError(f"run_process failed: {stderr_msg}")
```
Comment (SimonKrughoff) [inline tests/test_cli.py:131]: It seems to me like you would only want to raise if you get an `Exception`.  This appears to raise any time the code gets to the `else` block.

**Review by SimonKrughoff — APPROVED** [review]
I only have one substantial comment.  Looks great.

File: tests/test_cli.py (hunk 1 above)
Comment (r-owen) [reply tests/test_cli.py:80]: I am testing specifying a port and not specifying a port -- in other words, whether `explog run` and `explog run --port <port>` both work. 8080 is the default port when using `explog run`.

8000 could be any value. I probably should have picked something that looks less like 8080. It happens to be the default port when running using `tox -e run`. I have no idea how the template generator came up with 8080 and 8000 came from, nor why they differ between the two ways of running the service.

Comment (r-owen) [reply tests/test_cli.py:80]: I'll improve the doc string and add a code comment to explain what's going on.

File: tests/test_cli.py (hunk 2 above)
Comment (r-owen) [reply tests/test_cli.py:133]: This test expects that the `explog run` process is still be running at this point in the code. That's the if condition.
If the process has exited (the `else` condition) then the test fails. I make a "best effort" attempt to stderr from the process and include it in the error message, but I am going to report the failure with or without that information.

I'll add a code comment explaining.

Comment (SimonKrughoff) [reply tests/test_cli.py:139]: Thank you.  Sounds good.

**Pull request merged**